tracer/src/Datadog.Trace/Ci/TestSession.cs

```diff
@@ -102,21 +102,26 @@ private TestSession(string? command, string? workingDirectory, string? framework
             OpenedTestSessions.Add(this);
         }
 
-        _testOptimization.Log.Debug("### Test Session Created: {Command}", command);
-
-        if (startDate is null)
-        {
-            // If a module doesn't have a fixed start time we reset it before running code
-            span.ResetStartTime();
-        }
-
         // Record EventCreate telemetry metric
         if (TelemetryHelper.GetEventTypeWithCodeOwnerAndSupportedCiAndBenchmark(
                 MetricTags.CIVisibilityTestingEventType.Session,
                 framework == CommonTags.TestingFrameworkNameBenchmarkDotNet) is { } eventTypeWithMetadata)
         {
             TelemetryFactory.Metrics.RecordCountCIVisibilityEventCreated(TelemetryHelper.GetTelemetryTestingFrameworkEnum(framework), eventTypeWithMetadata);
         }
+
+        TelemetryFactory.Metrics.RecordCountCIVisibilityTestSession(
+            _testOptimization.CIValues.MetricTag,
+            EnvironmentHelpers.GetEnvironmentVariable(TestSuiteVisibilityTags.TestSessionAutoInjectedEnvironmentVariable)?.ToBoolean() is true ? MetricTags.CIVisibilityTestSessionType.AutoInjected : MetricTags.CIVisibilityTestSessionType.NotAutoInjected,
```

Comment:
nit, this is pretty hard to read IMO, extracting first param prob makes sense?
```suggestion
        var sessionTypeTag = EnvironmentHelpers.GetEnvironmentVariable(TestSuiteVisibilityTags.TestSessionAutoInjectedEnvironmentVariable)?.ToBoolean() is true 
            ? MetricTags.CIVisibilityTestSessionType.AutoInjected
            : MetricTags.CIVisibilityTestSessionType.NotAutoInjected;

        TelemetryFactory.Metrics.RecordCountCIVisibilityTestSession(
            _testOptimization.CIValues.MetricTag,
            sessionTypeTag,
```

Reply:
done in [86907ce](https://github.com/DataDog/dd-trace-dotnet/pull/6776/commits/86907ce60502094c1834c531fad1d579bf0f1169)